### PR TITLE
fix: Raise error when registering Packet-keyed converter

### PR DESCRIPTION
**Pull request by gs-olive**
# Description

- Packet-keyed converters are too error-prone, since new overloads could be added or existing overloads may have been overlooked

## Type of change

- Improved error-checking

# Checklist:

- [ x ] My code follows the style guidelines of this project (You can use the linters)
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ x ] I have made corresponding changes to the documentation
- [ - ] I have added tests to verify my fix or my feature
  - CI-validated
- [ x ] New and existing unit tests pass locally with my changes
- [ x ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: tests/py/dynamo/backend/test_specialized_models.py [hunk 1]
```diff
@@ -157,5 +157,41 @@ def forward(self, x):
...
+    def test_packet_operator(self):
+        class PacketAsOperator(torch.nn.Module):
+            def forward(self, x):
+                return torch.ops.aten.tanh(x)
+
+        # Operations expected to be removed in the traced graph
+        expected_ops = {torch.ops.aten.tanh.default}
+        unexpected_ops = {
+            torch.ops.aten.tanh,
+        }
```
Comment (gs-olive) [inline tests/py/dynamo/backend/test_specialized_models.py:170]: Validates that the Packet callable `torch.ops.aten.tanh` is transformed into `torch.ops.aten.tanh.default` by AOT autograd.

File: py/torch_tensorrt/dynamo/conversion/converter_registry.py [hunk 2]
```diff
@@ -101,6 +102,19 @@ def register_converter(converter: ConverterImplSignature) -> ConverterImplSignat
                 capability_validator=capability_validator,
             )
 
+        # OpOverloads are only valid if they have a single overload, or
```
Comment (apbose) [inline py/torch_tensorrt/dynamo/conversion/converter_registry.py:105]: Should the comment here be OpOverloadPacket
